### ISSUE TITLE
Blank page when rendering `ContextProvider` component

### ISSUE THREAD

**Issue by sdcdsvsdv423** [ryansolid/mobx-jsx#12]
**Reproduce:**
This error occurs [here](https://codesandbox.io/s/mobx-counter-context-wlu1x) if replace 
```js
const App = () => (
  <CounterProvider count={7}>
    <MiddleComponent />
  </CounterProvider>
);

render(() => <App/>, document.getElementById("app"));
```
to 
```js
render(() => App, document.getElementById("app"));
```
But the following code is working as expected:
```js
const App = () => (
  <div>
    <MiddleComponent />
  </div>
);

render(() => App, document.getElementById("app"));
```

**Comment by ryansolid** [comment]
Yeah this is similar to the other issue you were having. Since the reactive system treats functions as reactive and Components are functions in simple cases passing the Component declaration can work. However Component instances are different. When you wrap in JSX you are effectively doing this `App(props);`. Creating a Component instance does a few other things like dynamic prop handling. Now top level Component doesn't need that with no props so in theory we should be able to pass just the function. However passing a function that returns a function has different meaning. That will make the whole expression reactive. In this case a dynamic Component like Provider is likely returning a function as well double nesting reactivity. I'm not sure why in this case it stops working and I suppose I can investigate, but in no scenario is that preferable way to approach.

Basically the second case works incidentally but the correct form is either `render(() => <App/>, ...)` or `render(App, ...)` which are roughly equivalent. However  `render(() => App, ...)` is not. I do think there may be a bug buried in here somewhere so I will look into it. But I suspect it is MobX's reactive scheduling balking at double nesting which other libraries allow, but maybe they do not. However I suspect this avoidable if we only try to insert substantiated Components, ie make sure the Component ends up being wrapped in JSX before it ends up being passed to render or in an expression that is inserted in real DOM elements. That is how it would work in React. If you tried to just pass the function it would not behave as expected.

**Comment by sdcdsvsdv423** [comment]
It would be nice if the library gives some hint to prevent double reactive. The same problem appeared when there is nested context. Should be nested contexts supported?

**Comment by ryansolid** [comment]
To be fair, MobX is the reactive library I'm least familiar with that I've used with my renderer. I probably need to spend some more time with MobX specifically for non-trivial cases. Until the first issue you reported I wasn't aware since it wasn't something I'd try. I'm hoping with a bit of investigation I will figure it out. But it is possible double nesting reactivity might not work here which require rethinking some part of the solution.

Most of these solutions I approach from a fairly general reactive approach and do most of my non-trivial work with Solid before back porting into s-js, ko-jsx, and mobx-jsx. But on mobx-jsx specifically I don't usually spend that much time unless there are issues. So I hope to be able to say something definitive soon.

**Comment by ryansolid** [comment]
This should be fixed in v0.8.2.

**Comment by sdcdsvsdv423** [comment]
Thank you! I do not see blank page anymore.
